src/Plugin.php

```diff
@@ -35,9 +40,94 @@ class Plugin {
    * @implements init
    */
   public static function init() {
+    // Only users with an allowed role ('sale-editor' by default) or with the
+    // 'edit_product' capability can print the price label from the frontend.
+    if (!static::checkUserCanPrintLabel()) {
+      return;
+    }
+
+    if (!Label::$pdfDocsFolderPath || !Label::$pdfDocsFolderUrl) {
+      $uploads_dir = wp_get_upload_dir();
+      Label::$pdfDocsFolderPath = $uploads_dir['basedir'] . '/' . Plugin::PREFIX;
+      Label::$pdfDocsFolderUrl = $uploads_dir['baseurl'] . '/' . Plugin::PREFIX;
+    }
+
+    // Adds price label printing controls to product summary.
+    add_action('woocommerce_simple_add_to_cart', __CLASS__ . '::addPrintPriceLabelControls');
+    add_action('woocommerce_single_variation', __CLASS__ . '::addPrintPriceLabelControls');
+
+    // Prints product price label PDF document.
+    add_action('post_action_label', __NAMESPACE__ . '\Label::post_action_label');
+
+    // Schedule the cleanup of expired products price labels.
+    add_action(Plugin::PREFIX . '_delete_old_price_labels', __NAMESPACE__ . '\Label::deleteOldPriceLabels');
+
     if (is_admin()) {
       return;
     }
+
+    // Enqueues frontend plugin scripts and styles.
+    add_action('wp_enqueue_scripts', __CLASS__ . '::wp_enqueue_scripts');
+  }
+
+  /**
+   * Checks if user is allowed to print product price labels from frontend.
+   *
+   * @return bool
+   *   TRUE if the user can print product price labels from frontend.
+   */
+  public static function checkUserCanPrintLabel() {
```

Comment:
Doesn't it makes more sense to check (and possibly add) for a capability? In case a new role is introduced this needs to be added as well. Also please note there is the [`current_user_can`](https://developer.wordpress.org/reference/functions/current_user_can/) function which makes `wp_get_current_user` kind of obsolete.

Reply:
Yes, I do agree with you. My initial approach was to check for a capability, but @miguel-lucas asked me to check for a given role (sale-editor) instead.

Reply:
You could add the capability to roles and then check for the capability. Miguel won't notice the difference 🙃

Reply:
Ok, I will do that. Don't tell Miguel then 😉